src/installers/darwin/dmg.js

```diff
@@ -1,5 +1,31 @@
-import opn from 'opn';
+import { spawn } from 'child_process';
+import fs from 'fs-promise';
+import path from 'path';
 
-export default async (filePath) => {
-  await opn(filePath, { wait: false });
+import { getMountedImages, mountImage, unmountImage } from '../../util/hdiutil';
+import moveApp from '../../util/move-app';
+
+export default async (filePath, installSpinner) => {
+  const mounts = await getMountedImages();
+  let targetMount = mounts.find(mount => mount.imagePath === filePath);
+
+  if (!targetMount) {
+    targetMount = await mountImage(filePath);
+  }
+
+  const volumePath = path.resolve('/Volumes', targetMount.mountPath);
+  const appName = (await fs.readdir(volumePath)).find(file => file.endsWith('.app'));
+  if (!appName) {
+    await unmountImage(targetMount);
+    // eslint-disable-next-line no-throw-literal
```

Comment:
Is there a reason that in another PR we can't just disable `no-throw-literal` in the ESLint config? I count 18 instances of this same comment in master.